src/ui/components/TestSuite/utils/cypressStepDomNodes.ts

```diff
@@ -0,0 +1,119 @@
+declare var __RECORD_REPLAY_ARGUMENTS__: {
+  getPersistentId?: (value: any) => number | null | void;
+
+  internal?: {
+    registerPlainObject?: (value: any) => string | null | void;
+  };
+};
+
+declare var __RECORD_REPLAY__: {
+  getObjectFromProtocolId?: (id: string | number) => any;
+  getProtocolIdForObject?: (object: any) => string | number | null;
+};
+
+export function findDOMNodeObjectIdForPersistentId(persistentId: number): string | null {
+  if (
+    typeof __RECORD_REPLAY__ === "undefined" ||
+    __RECORD_REPLAY__ == null ||
+    __RECORD_REPLAY__.getProtocolIdForObject == null
+  ) {
+    return null;
+  }
+
+  if (
+    typeof __RECORD_REPLAY_ARGUMENTS__ === "undefined" ||
+    __RECORD_REPLAY_ARGUMENTS__ == null ||
+    __RECORD_REPLAY_ARGUMENTS__.getPersistentId == null
+  ) {
+    return null;
+  }
+
+  const { getPersistentId } = __RECORD_REPLAY_ARGUMENTS__;
+
+  let domNodeId: string | null = null;
+
+  function getObjectId(value: any) {
+    if (typeof __RECORD_REPLAY_ARGUMENTS__ !== "undefined" && __RECORD_REPLAY_ARGUMENTS__ != null) {
+      // TODO [FE-2005][FE-2067] With persistent DOM ids, presumably we should switch to using this API?
+      // if (__RECORD_REPLAY_ARGUMENTS__.getPersistentId) {
+      //   const id = __RECORD_REPLAY_ARGUMENTS__.getPersistentId(value);
+      //   if (id) {
+      //     return id;
+      //   }
+      // }
```

Comment:
This comment isn't relevant here, I think. We already have a handle on the DOM element (and its persistent id). That's what we're using to match up with the regular `ObjectId`